docker.go

```diff
@@ -138,6 +140,16 @@ func (r DockerContainerRun) WithEntrypoint(entrypoint string) DockerContainerRun
 	return r
 }
 
+func (r DockerContainerRun) WithPublishAll(port string) DockerContainerRun {
+	r.publishAllPorts = port
+	return r
+}
+
+func (r DockerContainerRun) WithPublish(hostPort, containerPort string) DockerContainerRun {
```

Comment:
I don't think we want to allow for specifying the host port here. I think the expectation should be that the user expose the container port and then use `WithPublishAll` to bind that port to a random port on the host. The reason we'd care about this is that we want the tests to be able to be run easily in parallel. If you have folks pick the same host port for different tests, then those ports might collide during a test run.

But, I think maybe it makes sense to just have a single string parameter like
```go
func (r DockerContainerRun) WithPublish(value string) DockerContainerRun
```
This way users could call this option like `WithPublish("8080:1234")` or `WithPublish("9292")`. It supports what you've implemented here and also the other case where the user is trying to bind to a random host port.

Reply:
Ah I was confused about this, that makes a lot more sense to do

Reply:
@ryanmoran do you think we should allow users to set multiple port mappings?

Reply:
Yeah. Docker allows it, so I think we should too.